src/main/java/org/veupathdb/service/eda/ds/plugin/pass/TablePlugin.java

```diff
@@ -101,10 +105,30 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
     
     //loop through and print data
     first = true;
+    int offsetCount = 0;
+    int rowCount = 0;
     while (s.hasNextLine()) {
-      String[] row = parser.parseLineToArray(s.nextLine());
-      if (first) first = false; else out.write(",".getBytes());
-      out.write(new JSONArray(row).toString().getBytes());
+      if (offset != null) {
+        if (offsetCount < offset) {
+          offsetCount++;
+          s.nextLine();
+        } else {
+          if (numRows != null) {
+            if (rowCount < numRows) {
+              rowCount++;
+              String[] row = parser.parseLineToArray(s.nextLine());
```

Comment:
Maybe break these three lines into a function?  Repetition.

Reply:
i agree its a bit repetitive.. and maybe im just being thick, but i dont really get what you mean by baking conditional logic into a function?